src/App.tsx

```diff
@@ -1,24 +1,254 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, {
+  useState,
+  useCallback,
+  useEffect,
+  useMemo,
+} from 'react';
+
+// components
 import { UserWarning } from './UserWarning';
+import { Header } from './components/Header';
+import { TodoList } from './components/TodoList';
+import { Footer } from './components/Footer/Footer';
+import { ErrorMessage } from './components/ErrorMessage';
+
+// utils for loading and updating
+import {
+  getTodos,
+  addTodo,
+  deleteTodo,
+  updateStatusTodo,
+  updateTitleTodo,
+} from './api/todos';
+
+// utils
+import { getFilteredTodos } from './utils/getFilteredTodos';
 
-const USER_ID = 0;
+// types
+import { Todo } from './types/Todo';
+import { FilterType } from './types/FilterType';
+import { ErrorMessages } from './types/ErrorMessages';
+
+const USER_ID = 6354;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [todosWithLoader, setTodosWithLoader] = useState<Todo[]>([]);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+
+  const [isLoadingFailed, setIsLoadingFailed] = useState(false);
+  const [errorMessage, setErrorMessage] = useState(ErrorMessages.NONE);
+
+  const [filterBy, setFilterBy] = useState<FilterType>(FilterType.ALL);
+
+  const visibleTodos = useMemo(() => (
+    getFilteredTodos(todos, filterBy)
+  ), [todos]);
+
+  const completedTodos = useMemo(() => (
+    todos.filter(todo => todo.completed === true)
+  ), [todos]);
+
+  const numberOfNotCompletedTodos = todos.length - completedTodos.length;
+  const isClearAllButtonVisible = Boolean(!completedTodos.length);
+  const isToogleButtonVisible = Boolean(!numberOfNotCompletedTodos);
+  const isSubmitButtonDisabled = Boolean(tempTodo?.title.length);
+
+  const addErrorMessage = useCallback((newMessage: ErrorMessages) => {
+    setErrorMessage(newMessage);
+  }, []);
+
+  const getAllTodos = useCallback(async () => {
+    try {
+      const allTodos = await getTodos(USER_ID);
+
+      setIsLoadingFailed(false);
+      setTodos(allTodos);
+    } catch {
+      setIsLoadingFailed(true);
+      addErrorMessage(ErrorMessages.LOAD);
+    }
+  }, []);
+
+  useEffect(() => {
+    getAllTodos();
+  }, []);
+
+  const clearMessage = useCallback(() => {
+    setErrorMessage(ErrorMessages.NONE);
+  }, [errorMessage]);
+
+  const addNewTodo = useCallback(async (title: string) => {
+    clearMessage();
+
+    if (!title) {
+      addErrorMessage(ErrorMessages.TITLE);
+
+      return;
+    }
+
+    const newTodo = {
+      id: 0,
+      userId: USER_ID,
+      title,
+      completed: false,
+    };
+
+    setTempTodo(newTodo);
+
+    try {
+      await addTodo(USER_ID, newTodo);
+      await getAllTodos();
```

Comment:
Id for new todo will be set on the server, so you shouldn't pass newTodo with id  to `addTodo`